src/providers/npm.ts

```diff
@@ -0,0 +1,203 @@
+import type { FontFaceData, FontFaceMeta, ResolveFontOptions } from '../types'
+import { existsSync } from 'node:fs'
+import { readFile } from 'node:fs/promises'
+import { join, resolve } from 'node:path'
+import { extractFontFaceData } from '../css/parse'
+import { defineFontProvider } from '../utils'
+
+interface NPMProviderOptions {
+  /**
+   * List of npm packages to check for fonts
+   * For example: ['@fontsource/poppins', '@fontsource/roboto']
+   * If not provided, will try to autodetect @fontsource packages
+   */
+  packages?: string[]
+
+  /**
+   * Base directory for resolving npm packages
+   * @default 'node_modules'
+   */
+  baseDir?: string
+
+  /**
+   * List of known font package prefixes to try
+   * @default ['@fontsource/']
+   */
+  packagePrefixes?: string[]
+}
+
+// Extend the FontFaceMeta interface
+interface NPMFontFaceMeta {
+  source: string
+  package: string
+}
+
+/**
+ * Try to detect installed @fontsource packages
+ */
+async function detectFontsourcePackages(baseDir: string): Promise<string[]> {
+  try {
+    // Check if @fontsource directory exists
+    const fontsourceDir = join(baseDir, '@fontsource')
+    if (!existsSync(fontsourceDir)) {
+      return []
+    }
+
+    // Read the directory to find installed font packages
+    const { readdir } = await import('node:fs/promises')
+    const files = await readdir(fontsourceDir)
+
+    return files.map(file => `@fontsource/${file}`)
+  }
+  catch (error) {
+    // In case of any error, return empty array
+    console.error('Failed to detect @fontsource packages:', error)
+    return []
+  }
+}
+
```

Comment:
let's instead try to check the `package.json` of the workspace - we can use `pkg-types` for this. And ideally I think we can also support other common fonts like `cal-sans`, geist, etc.